sdk/identity/identity/src/credentials/managedIdentityCredential/fabricMsi.ts

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AccessToken, GetTokenOptions, RequestPrepareOptions } from "@azure/core-http";
+import { MSI } from "./models";
+import { credentialLogger } from "../../util/logging";
+import { IdentityClient } from "../../client/identityClient";
+import { msiGenericGetToken } from "./utils";
+
+const logger = credentialLogger("ManagedIdentityCredential - Fabric MSI");
+
+function expiresInParser(requestBody: any): number {
+  // Parses a string representation of the seconds since epoch into a number value
+  return Number(requestBody.expires_on);
+}
+
+function prepareRequestOptions(resource: string, clientId?: string): RequestPrepareOptions {
+  const queryParameters: any = {
+    resource,
+    "api-version": "2019-07-01"
+  };
+
+  if (clientId) {
+    queryParameters.client_id = clientId;
+  }
+
+  return {
+    url: process.env.IDENTITY_ENDPOINT,
+    method: "GET",
+    queryParameters,
+    headers: {
+      Accept: "application/json",
+      Secret: process.env.IDENTITY_HEADER
+    }
+  };
+}
+
+// This credential can be easily tested by deploying a container to Azure Fabric with the Dockerfile:
```

Comment:
nit (though this comment will be updated when a manual test can target the JS credential):
```suggestion
// This credential can be easily tested by deploying a container to Azure Service Fabric with the Dockerfile:
```